File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/ChatIdResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.params.ChatId;

import java.lang.reflect.Parameter;

@Component
public class ChatIdResolver implements ParameterResolver<Long> {

    @Override
    public boolean supports(Parameter p) {
        return p.isAnnotationPresent(ChatId.class);
    }

    @Override
    public Long resolve(Parameter parameter, HandlerCtx ctx) {
        return ctx.chat().id();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/ChatIdResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.params.ChatId
import java.lang.reflect.Parameter

@Component
class ChatIdResolver : ParameterResolver<Long?> {
    override fun supports(p: Parameter?): Boolean? = p?.isAnnotationPresent(ChatId::class.java)

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): Long? = ctx.chat!!.id()

}
